src/Logger.cpp

```diff
@@ -86,7 +91,16 @@ void cLogger::AttachListener(cListener * a_Listener)
 void cLogger::DetachListener(cListener * a_Listener)
 {
 	cCSLock Lock(m_CriticalSection);
-	m_LogListeners.erase(std::remove(m_LogListeners.begin(), m_LogListeners.end(), a_Listener));
+	m_LogListeners.erase(
+		std::remove_if(
+			m_LogListeners.begin(),
+			m_LogListeners.end(),
+			[=](std::unique_ptr<cListener> & a_OtherListener) -> bool
+			{
+				return a_OtherListener.get() == a_Listener;
+			}
+		)
```

Comment:
How strange. When I was doing #2463, I tried this with a std::list<>::erase and ...vector... and both crashed if remove_if found nothing to remove. Yet it works here :/

Also, is it necessary to specify -> bool and must you capture everything by value? :P
Why not just put a_Listener in the capture list?